### DISCUSSION TITLE
Folder level auth tokens not clearing properly?

### DISCUSSION THREAD

**Discussion by bjornhansen** [Kong/insomnia#7778] [Help]
I haven't found reports on this behavior anywhere else, but I set up the new folder-level auth with Oauth 2.0 for my project. It worked great last week, but this week it's only using old tokens in the endpoints.

I've tried using the "Clear Oauth 2 session" button, and re-fetching the tokens in the refresh token area. This succeeds and displays new tokens in the folder auth panel, but the auth token inherited by the endpoints is still using some other token that has long since expired.

Am I doing something wrong or is this a bug? Thanks!

**Comment by subnetmarco** [comment]
@gatzjames is this something we addressed in the last beta?

**Reply by bjornhansen** [reply]
I realized that the "Clear Oauth 2 session" button does actually seem to work, I was just getting confusing token errors. Apologies for the false alarm.

But it would be awesome to have it clear the tokens on switching environments or remember the tokens on an environment basis.

**Comment by protono** [comment]
Hey. I have two requests r1 and r2 in a folder and they inherit oauth2 from that folder. I have two environments e1 and e2 that define oauth2. I want to run r1 with e1, switch to e2, clear tokens, fetch new tokens and run r1 with e2. That fails as it is still using e2 creds. I even exit Inso and 'Clear oauth2 on start' is switched on. After restarting Inso I can run r2 with e2 just fine, but r1 is stuck with e1. The only workaround I found is a copy/paste of r1 and that picks up e2 just fine. Is there a better way?
